detectable/src/main/java/com/synopsys/integration/detectable/detectables/bazel/BazelExtractor.java

```diff
@@ -1,72 +1,95 @@
+        this.dependencyTransformer = dependencyTransformer;
     }
 
-    public Extraction extract(ExecutableTarget bazelExe, File workspaceDir, BazelWorkspace bazelWorkspace, String bazelTarget,
-        BazelProjectNameGenerator bazelProjectNameGenerator, Set<WorkspaceRule> providedDependencyRuleTypes,
-        List<String> providedCqueryAdditionalOptions) {
-        logger.debug("Bazel extraction:");
+    public Extraction extract(ExecutableTarget bazelExe, File workspaceDir, File workspaceFile,
+        BazelProjectNameGenerator bazelProjectNameGenerator) throws DetectableException, ExecutableFailedException {
+        toolVersionLogger.log(workspaceDir, bazelExe, "version");
+        BazelCommandExecutor bazelCommandExecutor = new BazelCommandExecutor(executableRunner, workspaceDir, bazelExe);
+        Pipelines pipelines = new Pipelines(bazelCommandExecutor, bazelVariableSubstitutor, externalIdFactory, haskellCabalLibraryJsonProtoParser);
+        Set<WorkspaceRule> workspaceRulesFromFile = parseWorkspaceRulesFromFile(workspaceFile);
+        Set<WorkspaceRule> workspaceRulesToQuery = workspaceRuleChooser.choose(workspaceRulesFromFile, providedDependencyRuleTypes);
+        List<Dependency> aggregatedDependencies = collectDependencies(pipelines, workspaceRulesToQuery);
+        CodeLocation codeLocation = dependencyTransformer.createCodeLocation(aggregatedDependencies);
+        return buildResults(codeLocation, bazelProjectNameGenerator.generateFromBazelTarget(bazelTarget));
+    }
+
+    private Set<WorkspaceRule> parseWorkspaceRulesFromFile(final File workspaceFile) {
+        List<String> workspaceFileLines;
         try {
-            toolVersionLogger.log(workspaceDir, bazelExe, "version");
-            BazelCommandExecutor bazelCommandExecutor = new BazelCommandExecutor(executableRunner, workspaceDir, bazelExe);
-            BazelVariableSubstitutor bazelVariableSubstitutor = new BazelVariableSubstitutor(bazelTarget, providedCqueryAdditionalOptions);
-            Pipelines pipelines = new Pipelines(bazelCommandExecutor, bazelVariableSubstitutor, externalIdFactory);
-            Set<WorkspaceRule> workspaceRulesToQuery = workspaceRuleChooser.choose(bazelWorkspace.getDependencyRuleTypes(), providedDependencyRuleTypes);
-            List<Dependency> aggregatedDependencies = collectDependencies(pipelines, workspaceRulesToQuery);
-            return buildResults(aggregatedDependencies, bazelProjectNameGenerator.generateFromBazelTarget(bazelTarget));
-        } catch (Exception e) {
-            String msg = String.format("Bazel processing exception: %s", e.getMessage());
-            logger.debug(msg, e);
-            return new Extraction.Builder().failure(msg).build();
+            workspaceFileLines = FileUtils.readLines(workspaceFile, StandardCharsets.UTF_8);
+            return bazelWorkspaceFileParser.parseWorkspaceRuleTypes(workspaceFileLines);
+        } catch (IOException e) {
+            return new HashSet<>(0);
```

Comment:
We don't want any indication that we failed to parse this file?

Reply:
Good point. I've added a warning.